### ISSUE TITLE
network activity graph is -blank- in Debian Bullseye Mate

### ISSUE THREAD

**Issue by mexsudo** [jxai/lean-conky-config#2]
new glitch: this network activity graph is -blank- in Bullseye.
the text is still active...

![the missing graph 2021-05-15 08-06-48](https://user-images.githubusercontent.com/7226539/118368066-150e3680-b556-11eb-80ff-8864e2f0d191.png)

this screenshot is from one of several Debian 10 systems, I like it.
I have tried several fresh installs of Debian 11 in several machines and get the same result in all. 
I tried it with no packages installed but Conky, and then I did my best to duplicate the installed packages on the other Debian 10 systems.
(Debian 11 aka Bullseye.. still in Testing mode).

**Comment by jxai** [comment]
On Debian 11, what does the system return when running the following in command line:
```bash
ls -l /sys/class/net/*
```

**Comment by mexsudo** [comment]
peter@z820-2rc1:~$ ls -l /sys/class/net/*
lrwxrwxrwx 1 root root 0 May 15 20:26 /sys/class/net/eno1 -> ../../devices/pci0000:00/0000:00:19.0/net/eno1
lrwxrwxrwx 1 root root 0 May 15 20:26 /sys/class/net/enp1s0 -> ../../devices/pci0000:00/0000:00:1c.0/0000:01:00.0/net/enp1s0
lrwxrwxrwx 1 root root 0 May 15 20:26 /sys/class/net/lo -> ../../devices/virtual/net/lo
peter@z820-2rc1:~$

**Comment by mexsudo** [comment]
root@z820-2rc1:~# ls -l /sys/class/net/*
lrwxrwxrwx 1 root root 0 May 15 20:26 /sys/class/net/eno1 -> ../../devices/pci0000:00/0000:00:19.0/net/eno1
lrwxrwxrwx 1 root root 0 May 15 20:26 /sys/class/net/enp1s0 -> ../../devices/pci0000:00/0000:00:1c.0/0000:01:00.0/net/enp1s0
lrwxrwxrwx 1 root root 0 May 15 20:26 /sys/class/net/lo -> ../../devices/virtual/net/lo
root@z820-2rc1:~#

**Comment by jxai** [comment]
Devices seem ok, please post a screenshot from Debian 11. Also run conky from command line, and see if there are informative error messages:
```bash
conky -c /path/to/lean-conky-config/conky.conf
```

**Comment by mexsudo** [comment]
Debian 11
peter@z820-2rc1:~$ conky -c /home/peter/.config/conky/lean-conky-config-master/conky.conf
conky: warning: invalid head index, ignoring head settings
conky: warning: invalid head index, ignoring head settings
conky: desktop window (e00006) is subwindow of root window (5ba)
conky: window type - normal
conky: drawing to created window (0x5400002)
conky: drawing to double buffer

![d11lean-conky](https://user-images.githubusercontent.com/7226539/118494498-13b84780-b6d7-11eb-9884-f34211d2d61d.png)

====
Debian 10
peter@z840d10:~$ conky -c /home/peter/.config/conky/lean-conky-config-master/conky.conf
conky: warning: invalid head index, ignoring head settings
conky: warning: invalid head index, ignoring head settings
conky: desktop window (1400006) is subwindow of root window (5c8)
conky: window type - normal
conky: drawing to created window (0x8000002)
conky: drawing to double buffer

**Comment by jxai** [comment]
@mexsudo LCC has no problem identifying your network interface (enp1s0), you might want to check this with Conky developers why upspeedgraph/downspeedgraph show no activity on Debian 11.

Or try another Conky config with network activity graphs, see if you see any activities.

**Comment by mexsudo** [comment]
thank you for your efforts.

Debian 11 is in the final stages of testing. I will sit and wait.
Maybe we will get lucky and the official release will be compatible.

**Comment by mexsudo** [comment]
FYI: https://sourceforge.net/p/conky/support-requests/41/

**Comment by jxai** [comment]
> FYI: https://sourceforge.net/p/conky/support-requests/41/

1. Conky development has moved here: https://github.com/brndnmtthws/conky
2. You might want to try a simple config first to see if the issue can be reproduced, e.g. the following:
```lua
conky.config = {
    alignment = 'top_left',
    background = false,
    border_width = 1,
    cpu_avg_samples = 2,
    default_color = 'white',
    default_outline_color = 'white',
    default_shade_color = 'white',
    draw_borders = false,
    draw_graph_borders = true,
    draw_outline = false,
    draw_shades = false,
    use_xft = true,
    font = 'DejaVu Sans Mono:size=12',
    gap_x = 5,
    gap_y = 60,
    minimum_height = 5,
    minimum_width = 5,
    net_avg_samples = 2,
    no_buffers = true,
    out_to_console = false,
    out_to_stderr = false,
    extra_newline = false,
    own_window = true,
    own_window_class = 'Conky',
    own_window_type = 'desktop',
    stippled_borders = 0,
    update_interval = 1.0,
    uppercase = false,
    use_spacer = 'none',
    show_graph_scale = false,
    show_graph_range = false
}

conky.text = [[
${scroll 16 $nodename - $sysname $kernel on $machine | }
$hr
${color grey}Uptime:$color $uptime
${color grey}Frequency (in MHz):$color $freq
${color grey}Frequency (in GHz):$color $freq_g
${color grey}RAM Usage:$color $mem/$memmax - $memperc% ${membar 4}
${color grey}Swap Usage:$color $swap/$swapmax - $swapperc% ${swapbar 4}
${color grey}CPU Usage:$color $cpu% ${cpubar 4}
${color grey}Processes:$color $processes  ${color grey}Running:$color $running_processes
$hr
${color grey}File systems:
 / $color${fs_used /}/${fs_size /} ${fs_bar 6 /}
$hr
${color grey}Networking:
Up:$color ${upspeed eth0} ${color grey} - Down:$color ${downspeed eth0}
${downspeedgraph eth0}
${upspeedgraph eth0}
$hr
${color grey}Name              PID   CPU%   MEM%
${color lightgrey} ${top name 1} ${top pid 1} ${top cpu 1} ${top mem 1}
${color lightgrey} ${top name 2} ${top pid 2} ${top cpu 2} ${top mem 2}
${color lightgrey} ${top name 3} ${top pid 3} ${top cpu 3} ${top mem 3}
${color lightgrey} ${top name 4} ${top pid 4} ${top cpu 4} ${top mem 4}
]]
```
Replace eth0 with the actual network interface you on your system, from your screenshot above it should be `enp1s0`.

**Comment by Tribleave** [comment]
I also found this problem in my arch linux. 
I think it is not caused by distribution's version, because it conky can detect my device `wlan0`.
I did some tests, the one is to directly show `upspeedgraph wlan0 32,130` in `conky.conf`  and it worked properly.
So I think this problem may caused by [conky ifaces](https://github.com/jxai/lean-conky-config/blob/1d10c8278fb4289ce8d81d9efb8fd8e8c5db0581/main.lua#L156).
